fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java

```diff
@@ -1670,8 +1670,10 @@ void buildPartitions(Database db, OlapTable table, List<Partition> partitions) t
         if (partitions.isEmpty()) {
             return;
         }
-        int numAliveCNs = Config.only_use_compute_node ? systemInfoService.getAliveComputeNodeNumber() :
-                systemInfoService.getAliveBackendNumber();
+        int numAliveCNs = systemInfoService.getAliveBackendNumber();
```

Comment:
looks like, the original `numBackends` name is more properly here. This code path is shared by both local table and cloudnative table. Local table can only use BE to build partition while the cloudnative table can also leverage CN to build partition.

So the main logic is: use BE to build partition, if is cloudnative table, add CN into resource pool as well.